src/apps/integrations/oneup_health/service/ehr_storage.py

```diff
@@ -0,0 +1,47 @@
+import json
+import uuid
+from io import BytesIO
+
+from apps.file.enums import FileScopeEnum
+from apps.file.storage import select_storage
+from apps.integrations.oneup_health.service.domain import EHRData
+from infrastructure.utility import CDNClient
+
+__all = ["create_ehr_storage"]
+
+
+class _EHRStorage:
+    def __init__(self, session, applet_id: uuid.UUID, cdn_client: CDNClient):
+        self._session = session
+        self._applet_id = applet_id
+        self._cdn_client: CDNClient = cdn_client
+
+    def _get_storage_path(self, base_path: str, key: str) -> str:
+        index = key.find(base_path)
+        if index == -1:  # substring not found
+            return key
+
+        # Return everything up to and including the unique substring
+        return key[: index + len(base_path)]
+
+    async def upload_resources(self, data: EHRData):
+        base_path = f"{data.unique_id}/{data.date.strftime('%Y-%m-%d')}"
+        filename = "resources.json"
```

Comment:
Adding a TODO as a reminder to update as part of a later ticket
```suggestion
        # TODO: Follow updated filename format [M2-8831](https://mindlogger.atlassian.net/browse/M2-8831)
        base_path = f"{data.unique_id}/{data.date.strftime('%Y-%m-%d')}"
        filename = "resources.json"
```